pinot-controller/src/main/java/org/apache/pinot/controller/helix/RealtimeConsumerMonitor.java

```diff
@@ -0,0 +1,122 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.helix;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import org.apache.commons.httpclient.SimpleHttpConnectionManager;
+import org.apache.pinot.common.metrics.ControllerGauge;
+import org.apache.pinot.common.metrics.ControllerMetrics;
+import org.apache.pinot.controller.ControllerConf;
+import org.apache.pinot.controller.LeadControllerManager;
+import org.apache.pinot.controller.helix.core.PinotHelixResourceManager;
+import org.apache.pinot.controller.helix.core.periodictask.ControllerPeriodicTask;
+import org.apache.pinot.controller.util.ConsumingSegmentInfoReader;
+import org.apache.pinot.spi.config.table.TableType;
+import org.apache.pinot.spi.stream.PartitionLagState;
+import org.apache.pinot.spi.utils.builder.TableNameBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class RealtimeConsumerMonitor extends ControllerPeriodicTask<RealtimeConsumerMonitor.Context> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(RealtimeConsumerMonitor.class);
+  private final ConsumingSegmentInfoReader _consumingSegmentInfoReader;
+
+  @VisibleForTesting
+  public RealtimeConsumerMonitor(ControllerConf controllerConf, PinotHelixResourceManager pinotHelixResourceManager,
+      LeadControllerManager leadControllerManager, ControllerMetrics controllerMetrics,
+      ConsumingSegmentInfoReader consumingSegmentInfoReader) {
+    super("RealtimeConsumerMonitor", controllerConf.getRealtimeConsumerMonitorRunFrequency(),
+        controllerConf.getRealtimeConsumerMonitorInitialDelayInSeconds(), pinotHelixResourceManager,
+        leadControllerManager, controllerMetrics);
+    _consumingSegmentInfoReader = consumingSegmentInfoReader;
+  }
+
+  public RealtimeConsumerMonitor(ControllerConf controllerConf, PinotHelixResourceManager pinotHelixResourceManager,
+      LeadControllerManager leadControllerManager, ControllerMetrics controllerMetrics,
+      ExecutorService executorService) {
+    this(controllerConf, pinotHelixResourceManager, leadControllerManager, controllerMetrics,
+        new ConsumingSegmentInfoReader(executorService, new SimpleHttpConnectionManager(), pinotHelixResourceManager));
+  }
+
+  @Override
+  protected void setUpTask() {
+    LOGGER.info("Setting up RealtimeConsumerMonitor task");
+  }
+
+  @Override
+  protected void processTable(String tableNameWithType) {
+    if (!TableType.REALTIME.equals(TableNameBuilder.getTableTypeFromTableName(tableNameWithType))) {
+      return;
+    }
+    try {
+      ConsumingSegmentInfoReader.ConsumingSegmentsInfoMap segmentsInfoMap =
+          _consumingSegmentInfoReader.getConsumingSegmentsInfo(tableNameWithType, 10000);
```

Comment:
It looks that `10000` is to configure the time out. Can we use the `static final int` variable for this? `e.g. DEFAULT_TIMEOUT_MS`